### ISSUE TITLE
plasma5.systemsettings can't load: kcm_splashscreen, kcm_desktoptheme, kcm_lookandfeel

### ISSUE THREAD

**Issue by zraexy** [NixOS/nixpkgs#27050]
## Issue description

System Settings cannot load `kcm_splashscreen`, `kcm_desktoptheme`, and `kcm_lookandfeel` when you go to their respective tabs.

Journal:
```
Jul 02 06:23:17 phoenix xsession[1533]: Error loading KCM plugin "kcm_splashscreen" "The shared library was not found."
Jul 02 06:23:17 phoenix xsession[1533]: The environment variable QT_PLUGIN_PATH might be not correctly set
Jul 02 06:23:17 phoenix xsession[1533]: Plugin search paths are ("/home/me/lib/qt-5.9/plugins", "/run/wrappers/lib/qt-5.9/plugins", "/etc/per-user-pkgs/me/lib/qt-5.9/plugins", "/home/me/.nix-profile/lib/qt-5.9/plugins", "/nix/var/nix/profiles/default/lib/qt-5.9/plugins", "/run/current-system/sw/lib/qt-5.9/plugins", "/nix/store/k7yy0jklv39y6zh62mix9x8l6x9m77lk-system-path/lib/kde4/plugins", "/nix/store/2jxzjifgc9if9lqrppg5k34qg89rwfvn-systemsettings-5.10.3-bin/bin")
Jul 02 06:23:17 phoenix xsession[1533]: org.kde.kcoreaddons: Error loading plugin "kcm_splashscreen" "The shared library was not found."
Jul 02 06:23:16 phoenix xsession[1533]: Error loading KCM plugin "kcm_desktoptheme" "The shared library was not found."
Jul 02 06:23:16 phoenix xsession[1533]: The environment variable QT_PLUGIN_PATH might be not correctly set
Jul 02 06:23:16 phoenix xsession[1533]: Plugin search paths are ("/home/me/lib/qt-5.9/plugins", "/run/wrappers/lib/qt-5.9/plugins", "/etc/per-user-pkgs/me/lib/qt-5.9/plugins", "/home/me/.nix-profile/lib/qt-5.9/plugins", "/nix/var/nix/profiles/default/lib/qt-5.9/plugins", "/run/current-system/sw/lib/qt-5.9/plugins", "/nix/store/k7yy0jklv39y6zh62mix9x8l6x9m77lk-system-path/lib/kde4/plugins", "/nix/store/2jxzjifgc9if9lqrppg5k34qg89rwfvn-systemsettings-5.10.3-bin/bin")
Jul 02 06:23:16 phoenix xsession[1533]: org.kde.kcoreaddons: Error loading plugin "kcm_desktoptheme" "The shared library was not found."
Jul 02 06:23:13 phoenix xsession[1533]: Error loading KCM plugin "kcm_lookandfeel" "The shared library was not found."
Jul 02 06:23:13 phoenix xsession[1533]: The environment variable QT_PLUGIN_PATH might be not correctly set
Jul 02 06:23:13 phoenix xsession[1533]: Plugin search paths are ("/home/me/lib/qt-5.9/plugins", "/run/wrappers/lib/qt-5.9/plugins", "/etc/per-user-pkgs/me/lib/qt-5.9/plugins", "/home/me/.nix-profile/lib/qt-5.9/plugins", "/nix/var/nix/profiles/default/lib/qt-5.9/plugins", "/run/current-system/sw/lib/qt-5.9/plugins", "/nix/store/k7yy0jklv39y6zh62mix9x8l6x9m77lk-system-path/lib/kde4/plugins", "/nix/store/2jxzjifgc9if9lqrppg5k34qg89rwfvn-systemsettings-5.10.3-bin/bin")
Jul 02 06:23:13 phoenix xsession[1533]: org.kde.kcoreaddons: Error loading plugin "kcm_lookandfeel" "The shared library was not found."
```

cc @ttuegel 
### Steps to reproduce

Have `services.xserver.desktopManager.plasma5.enable = true;` in `/etc/nixos/configuration.nix`.

`nixos-rebuild switch`

Open System Settings and click on `"Workspace Theme" > "Look And Feel"`, `"Workspace Theme" > "Desktop Theme"`, and `"Workspace Theme" > "Splash Screen"`.

## Technical details

* System: NixOS: 17.09pre110094.1add3519df (Hummingbird)
* Nix version: nix-env (Nix) 1.11.11
* Nixpkgs version: "17.09pre110094.1add3519df"
* Sandboxing enabled: build-use-sandbox = true

**Comment by matejc** [comment]
workaround
```
environment.variables.QT_PLUGIN_PATH = [ "${pkgs.plasma-desktop}/lib/qt-5.9/plugins/kcms" ];
```

**Comment by zraexy** [comment]
This was fixed a while ago. I'm not sure what commit it was though.

**Comment by hlandau** [comment]
This isn't fixed in 17.09. It also applies to kcm_pulseaudio.

**Comment by ttuegel** [comment]
I still cannot reproduce this issue.

@hlandau What does `nix-env -q` say, and what does `sudo nix-env -q` say?

**Comment by hlandau** [comment]
`nix-env -q` what, exactly?

The problem seems to be that `kcm_pulseaudio.so` is placed in /nix/store/...-system-path/lib/qt-5.9/plugins/kcms/kcm_pulseaudio.so - that is, in a `kcms` subdirectory rather than in `lib/qt-5.9/plugins`.

```
$ find /run/current-system/sw/lib/qt-5.9/plugins/ | grep kcm
/run/current-system/sw/lib/qt-5.9/plugins/kcm_view1394.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_kgamma.so
/run/current-system/sw/lib/qt-5.9/plugins/libkcm_device_automounter.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_icons.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_phonon.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_powerdevilglobalconfig.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_dolphinviewmodes.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_kwinoptions.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_memory.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_access.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_info.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_colors.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_filetypes.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_input.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_kded.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_devinfo.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_kwindesktop.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_kwinrules.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_kwintabbox.so
/run/current-system/sw/lib/qt-5.9/plugins/kcms
/run/current-system/sw/lib/qt-5.9/plugins/kcms/kcm_energyinfo.so
/run/current-system/sw/lib/qt-5.9/plugins/kcms/kcm_desktoptheme.so
/run/current-system/sw/lib/qt-5.9/plugins/kcms/kcm_pulseaudio.so
/run/current-system/sw/lib/qt-5.9/plugins/kcms/kcm_fileindexermonitor.so
/run/current-system/sw/lib/qt-5.9/plugins/kcms/kcm_lookandfeel.so
/run/current-system/sw/lib/qt-5.9/plugins/kcms/kcm_splashscreen.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_ssl.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_keys.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_smserver.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_webshortcuts.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_opengl.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_trash.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_cursortheme.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_kdegtkconfig.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_samba.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_style.so
/run/current-system/sw/lib/qt-5.9/plugins/libkcm_translations.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_krunner_dictionary.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_dolphinservices.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_componentchooser.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_printer_manager.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_kwallet5.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_nic.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_krunner_spellcheck.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_emoticons.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_formats.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_fonts.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_powerdevilprofilesconfig.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_fontinst.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_launch.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_knotify.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_pci.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_solid_actions.so
/run/current-system/sw/lib/qt-5.9/plugins/kcmspellchecking.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_dolphinnavigation.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_joystick.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_keyboard.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_kwindecoration.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_kio.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_powerdevilactivitiesconfig.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_plasmasearch.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_about_distro.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_dolphingeneral.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_autostart.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_activities.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_desktoppaths.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_krunner_kill.so
/run/current-system/sw/lib/qt-5.9/plugins/screenlocker_kcm.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_clock.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_kwinscreenedges.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_kwin_scripts.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_baloofile.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_kwintouchscreen.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_standard_actions.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_usb.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_kscreen.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_workspaceoptions.so
/run/current-system/sw/lib/qt-5.9/plugins/kcm_hotkeys.so
```

If I run systemsettings5 in a terminal and go to "Multimedia", the following output is generated:
```
org.kde.kcoreaddons: Error loading plugin "kcm_pulseaudio" "The shared library was not found." 
Plugin search paths are ("/home/hugo/lib/qt-5.9/plugins", "/run/wrappers/lib/qt-5.9/plugins", "/etc/per-user-pkgs/hugo/lib/qt-5.9/plugins", "/home/hugo/.nix-profile/lib/qt-5.9/plugins", "/nix/var/nix/profiles/default/lib/qt-5.9/plugins", "/run/current-system/sw/lib/qt-5.9/plugins", "/nix/store/waf806j0naskbxcpaaava03apr6dhi61-plasma-desktop-5.10.5/lib/qt-5.9/plugins/kcms", "/nix/store/gqgl27ycx7c9v0b8xfaq9kac5wch38rv-system-path/lib/kde4/plugins", "/nix/store/kd8jqai08pyw0w41czds99b5mzk0cg0n-systemsettings-5.10.5-bin/bin") 
The environment variable QT_PLUGIN_PATH might be not correctly set
Trying to use rootObject before initialization is completed, whilst using setInitializationDelayed. Forcing completion
```
Selecting the "audio and video" tab then generates the following additional output:
```
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
"/home/hugo/.config/kde.org/libphonon.conf"
libpaths ("/home/hugo/lib/qt-5.9/plugins", "/run/wrappers/lib/qt-5.9/plugins", "/etc/per-user-pkgs/hugo/lib/qt-5.9/plugins", "/home/hugo/.nix-profile/lib/qt-5.9/plugins", "/nix/var/nix/profiles/default/lib/qt-5.9/plugins", "/run/current-system/sw/lib/qt-5.9/plugins", "/nix/store/waf806j0naskbxcpaaava03apr6dhi61-plasma-desktop-5.10.5/lib/qt-5.9/plugins/kcms", "/nix/store/gqgl27ycx7c9v0b8xfaq9kac5wch38rv-system-path/lib/kde4/plugins", "/nix/store/kd8jqai08pyw0w41czds99b5mzk0cg0n-systemsettings-5.10.5-bin/bin")
void BackendSelection::load() "/home/hugo/lib/qt-5.9/plugins/phonon4qt5_backend" does not exist
void BackendSelection::load() "/run/wrappers/lib/qt-5.9/plugins/phonon4qt5_backend" does not exist
void BackendSelection::load() "/etc/per-user-pkgs/hugo/lib/qt-5.9/plugins/phonon4qt5_backend" does not exist
void BackendSelection::load() "/home/hugo/.nix-profile/lib/qt-5.9/plugins/phonon4qt5_backend" does not exist
void BackendSelection::load() "/nix/var/nix/profiles/default/lib/qt-5.9/plugins/phonon4qt5_backend" does not exist
void BackendSelection::load() "/run/current-system/sw/lib/qt-5.9/plugins/phonon4qt5_backend" does not exist
void BackendSelection::load() "/nix/store/waf806j0naskbxcpaaava03apr6dhi61-plasma-desktop-5.10.5/lib/qt-5.9/plugins/kcms/phonon4qt5_backend" does not exist
void BackendSelection::load() "/nix/store/gqgl27ycx7c9v0b8xfaq9kac5wch38rv-system-path/lib/kde4/plugins/phonon4qt5_backend" does not exist
void BackendSelection::load() "/nix/store/kd8jqai08pyw0w41czds99b5mzk0cg0n-systemsettings-5.10.5-bin/bin/phonon4qt5_backend" does not exist
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
"/home/hugo/.config/kde.org/libphonon.conf"
libpaths ("/home/hugo/lib/qt-5.9/plugins", "/run/wrappers/lib/qt-5.9/plugins", "/etc/per-user-pkgs/hugo/lib/qt-5.9/plugins", "/home/hugo/.nix-profile/lib/qt-5.9/plugins", "/nix/var/nix/profiles/default/lib/qt-5.9/plugins", "/run/current-system/sw/lib/qt-5.9/plugins", "/nix/store/waf806j0naskbxcpaaava03apr6dhi61-plasma-desktop-5.10.5/lib/qt-5.9/plugins/kcms", "/nix/store/gqgl27ycx7c9v0b8xfaq9kac5wch38rv-system-path/lib/kde4/plugins", "/nix/store/kd8jqai08pyw0w41czds99b5mzk0cg0n-systemsettings-5.10.5-bin/bin")
void BackendSelection::load() "/home/hugo/lib/qt-5.9/plugins/phonon4qt5_backend" does not exist
void BackendSelection::load() "/run/wrappers/lib/qt-5.9/plugins/phonon4qt5_backend" does not exist
void BackendSelection::load() "/etc/per-user-pkgs/hugo/lib/qt-5.9/plugins/phonon4qt5_backend" does not exist
void BackendSelection::load() "/home/hugo/.nix-profile/lib/qt-5.9/plugins/phonon4qt5_backend" does not exist
void BackendSelection::load() "/nix/var/nix/profiles/default/lib/qt-5.9/plugins/phonon4qt5_backend" does not exist
void BackendSelection::load() "/run/current-system/sw/lib/qt-5.9/plugins/phonon4qt5_backend" does not exist
void BackendSelection::load() "/nix/store/waf806j0naskbxcpaaava03apr6dhi61-plasma-desktop-5.10.5/lib/qt-5.9/plugins/kcms/phonon4qt5_backend" does not exist
void BackendSelection::load() "/nix/store/gqgl27ycx7c9v0b8xfaq9kac5wch38rv-system-path/lib/kde4/plugins/phonon4qt5_backend" does not exist
void BackendSelection::load() "/nix/store/kd8jqai08pyw0w41czds99b5mzk0cg0n-systemsettings-5.10.5-bin/bin/phonon4qt5_backend" does not exist
context_state_callback "Terminated"
context_state_callback "Terminated"
context_state_callback "Terminated"
Doing update 0
Got info about card "Built-in Audio"
Doing update 0
Got info about card "Built-in Audio"
Got info about sink "Built-in Audio Analog Stereo"
"Want ID: No; Have ID: No"
Got info about source "Built-in Audio Analog Stereo"
"Want ID: No; Have ID: No"
Doing update 0
```

**Comment by ttuegel** [comment]
> nix-env -q what, exactly?

Literally that command, `nix-env -q`. The packages installed in your user and system profile can interfere with plugin loading if they are outdated; that command lists them.

> that is, in a kcms subdirectory rather than in lib/qt-5.9/plugins.

That is where they are supposed to be. `kcmutils` searches the `kcms` subdirectory of the entries in the plugin path. The error message does not reflect the actual search path for KCMs. I patched `kcmutils` to output extra debugging information from the plugin loader; unfortunately now it prints some irrelevant stuff, too. (The original message doesn't even tell you _which_ plugin failed to load, it's totally useless for bug reports). The only accurate part of the message is
```
Error loading plugin "kcm_pulseaudio" "The shared library was not found."
```
Even this isn't _entirely_ true. We have to take it with a grain of salt: maybe it really didn't find the library, but maybe it found *a* library that it couldn't load. That's why I wanted to know what is in your user and system profile; to know if there is anything that could be interfering with loading.

If there are no outdated packages in your profiles, could you try to run `systemsettings5` under `strace`? That will confirm if the library truly cannot be found, or just cannot be loaded.

I should add: I do get the same message that you see, but then System Settings _loads the KCM anyway_ despite claiming it "can't be found"; the message itself seems to be spurious.

**Comment by hlandau** [comment]
`nix-env -q` and `sudo nix-env -q` don't output anything. I haven't installed any packages via `nix-env`.

**Comment by hlandau** [comment]
strace reveals that `kcm_pulseaudio` is being successfully opened. False alarm, I guess.

Here's what "Audio and Video" looks like. Is the pane supposed to be blank? https://i.imgur.com/bclcs61.png

**Comment by ttuegel** [comment]
> Here's what "Audio and Video" looks like. Is the pane supposed to be blank?

No, it shouldn't be, and I do see the same thing. It seems that at some point we stopped installing the Phonon Qt 5 backend and started installing the Qt 4 Phonon backend.

**Comment by hlandau** [comment]
https://github.com/NixOS/nixpkgs/commit/b25deccd1a75c3ab8e3cb4d085b62895e8e3c253 doesn't seem to fix the issue. Not sure if it's yet in 17.09, so I applied it manually and still have the same issue.

**Comment by peterchibunna** [comment]
`sudo apt install qml-module-org-kde-kcm` should fix it.

**Comment by benley** [comment]
@peterchibunna I mean this respectfully, but you are aware that this is the issue tracker for *NixOS* and not debian or ubuntu, yes?

**Comment by ttuegel** [comment]
> sudo apt install qml-module-org-kde-kcm should fix it.

That is unlikely to help on NixOS 😉

**Comment by ttuegel** [comment]
@hlandau What does `find /run/current-system/sw/lib -name '*phonon4qt5*'` say?

**Comment by cx405** [comment]
I am also affected by this in very fresh 17.09.
This is a basic stock desktop system. Installing kdeApplications.kmix, adding user to "audio" group didn't fix this. This leads to all KDE audio applications being unable to output sound or even to start.

@ttuegel  
`$ find /run/current-system/sw/lib -name '*phonon4qt5*'
/run/current-system/sw/lib/libphonon4qt5experimental.so.4.9.1
/run/current-system/sw/lib/libphonon4qt5.so.4.9.1
/run/current-system/sw/lib/libphonon4qt5.so.4
/run/current-system/sw/lib/libphonon4qt5experimental.so
/run/current-system/sw/lib/libphonon4qt5.so
/run/current-system/sw/lib/libphonon4qt5experimental.so.4`
>

**Comment by cx405** [comment]
When switching to empty tab of Phonon backend, I get this in terminal:
`WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
WARNING: bool Phonon::FactoryPrivate::createBackend() phonon backend plugin could not be loaded
"/home/userp/.config/kde.org/libphonon.conf"
libpaths ("/home/userp/lib/qt-5.9/plugins", "/run/wrappers/lib/qt-5.9/plugins", "/etc/per-user-pkgs/userp/lib/qt-5.9/plugins", "/home/userp/.nix-profile/lib/qt-5.9/plugins", "/nix/var/nix/profiles/default/lib/qt-5.9/plugins", "/run/current-system/sw/lib/qt-5.9/plugins", "/nix/store/ysch0j7fl06vir5a012nn92bav1qngdk-system-path/lib/kde4/plugins", "/nix/store/vg6g37dibbsgkhi1h1mxafmr2bj9skvj-systemsettings-5.10.5-bin/bin")
void BackendSelection::load() "/home/userp/lib/qt-5.9/plugins/phonon4qt5_backend" does not exist
void BackendSelection::load() "/run/wrappers/lib/qt-5.9/plugins/phonon4qt5_backend" does not exist
void BackendSelection::load() "/etc/per-user-pkgs/userp/lib/qt-5.9/plugins/phonon4qt5_backend" does not exist
void BackendSelection::load() "/home/userp/.nix-profile/lib/qt-5.9/plugins/phonon4qt5_backend" does not exist
void BackendSelection::load() "/nix/var/nix/profiles/default/lib/qt-5.9/plugins/phonon4qt5_backend" does not exist
void BackendSelection::load() "/run/current-system/sw/lib/qt-5.9/plugins/phonon4qt5_backend" does not exist
void BackendSelection::load() "/nix/store/ysch0j7fl06vir5a012nn92bav1qngdk-system-path/lib/kde4/plugins/phonon4qt5_backend" does not exist
void BackendSelection::load() "/nix/store/vg6g37dibbsgkhi1h1mxafmr2bj9skvj-systemsettings-5.10.5-bin/bin/phonon4qt5_backend" does not exist`

**Comment by cx405** [comment]
switching to unstable phonon and phonon-backend-gstreamer pulled unstable gstreamer packages, but the backend problem stayed same.

Members at irc suggested that it may be involved with hardcoded paths in backends: https://bugs.kde.org/show_bug.cgi?id=340126

**Comment by cx405** [comment]
My recent attempts still brought nothing:

nixpkgs.config.pulseaudio = true;

environment.systemPackages = with pkgs; [
....
pavucontrol
pamixer
phonon-backend-vlc
phonon-backend-gstreamer
... ];

hardware.pulseaudio.enable = true;
hardware.pulseaudio.package = pkgs.pulseaudioFull;
hardware.pulseaudio.support32Bit = true;
sound.enable = true;

The user is in audio group.

**Comment by cx405** [comment]
Ok, the user **ottidmes**  from #nixos IRC has recommended to add this inside systemPackages:

environment.systemPackages = with pkgs; [  
...  
(pkgs.runCommand "ln-phonon4qt5_backend" { } ''  
mkdir -p $out/lib/qt-5.9/plugins  
ln -s {/run/current-system/sw/lib/qt5,$out/lib/qt-5.9}/plugins/phonon4qt5_backend
'')  
...  
];

this has actually worked. Now I see two backends in settings and can also assign devices.
so the paths to backends are actually hardcoded.


I am really not in position to ask, but it would be so nice if there is a chance that you could push "proper" fix to this bug, @ttuegel . At least, newbies like me would seriously appreciate it!

**Comment by ttuegel** [comment]
> I am really not in position to ask, but it would be so nice if there is a chance that you could push "proper" fix to this bug, @ttuegel .

I'm afraid I don't know what the proper fix is because I still don't understand why the problem is. It appears as if your system is mixing Qt versions, but we fixed the problems associated with that before NixOS 17.09.

**Comment by cx405** [comment]
@ttuegel  I see. I will be doing 18.03 reinstallation today and will report how it went out, perhaps we can then decide how to fare from that point if the problem repeats.

**Comment by cx405** [comment]
@ttuegel  Hello again!

1. I have done complete new installation using 17.09 and doing channel update (still 17.09) before installation. I encountered #27307 and hence created a reduced configuration.nix. Installation was successful.  

After boot into the system, I could confirm this problem to exist for 17.09 stock.
This version did NOT contain the workaround for this bug in configuration.nix.  


2. Afterwards, I replaced configuration.nix with uncut version, replaced the "nixos" alias with current 18.03 channel and did "nixos-rebuild -Q test".  
After several hours due to qt5-webkit required to be built, the upgrade finished without error. I did "nixos-rebuild -Q switch" and rebooted. After reboot into 18.03, I couldn't reproduce the problem.  

This was also  WITHOUT workaround lines in configuration.nix.

3. Afterwards, I booted into 17.09 profile and can confirm that the bug still existed there.

Screenshots: https://imgur.com/a/WuidZ
Both configuration.nix: https://gist.github.com/cx405/d628b9c12b6f7b5f58053a2efb94dbe5

So, I can assume:
- this bug exists in current 17.09 amd64+kde image and its channels
- it can be fixed in 17.09 by adding these the lines, I have done many experimentation and reboots
- if the system upgrades to 18.03 using current 18.03 channel, it is fixed

Thus somewhere in 18.03 this bug was fixed in the newer package version, however it went unfixed through 17.09 (including after upgrading with its current channel contents).

I have not yet done "nix-collect-garbage -d", but I think this will change nothing.

Now, I don't know NixOS policy, so I can't answer for 17.09 - if this should be patched and backported to 17.09 or left broken there requiring manual fix. For 18.03, no fix seems to be required.

**Comment by ttuegel** [comment]
@cx405 In your NixOS 17.09 profile, is there anything in `/run/current-system/sw/lib/qt5`? It should be empty. If it is not empty, is there anything in `/run/current-system/sw/lib/qt-5.9`?

**Comment by cx405** [comment]
I will be doing new 17.09 install today and will report the contents. Thanks!

**Comment by cx405** [comment]
Well, it takes a while to get to new install as it seems. I am aware that I need to cooperate and aware of this bug, I will report back topmost within month.

**Comment by peterhoeg** [comment]
Is this still a problem?

**Comment by cx405** [comment]
@peterhoeg Sorry, I got dragged by other problems. This bug was fixed in 18.03.
